bricks/dart_frog_prod_server/hooks/src/create_external_packages_folder.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:io';
+
+import 'package:io/io.dart' as io;
+import 'package:path/path.dart' as path;
+import 'package:pubspec_lock/pubspec_lock.dart';
+
+Future<List<String>> createExternalPackagesFolder(
+  Directory directory, {
+  path.Context? pathContext,
+  Future<void> Function(String from, String to) copyPath = io.copyPath,
+}) async {
+  final pathResolver = pathContext ?? path.context;
+  final pubspecLock = await _getPubspecLock(directory.path, pathResolver);
+
+  final pathDependencies = pubspecLock.packages
+      .map(
+        (p) => p.iswitch(
+          sdk: (_) => null,
+          hosted: (_) => null,
+          git: (_) => null,
+          path: (d) => d.path,
+        ),
+      )
+      .whereType<String>()
+      .where((dependencyPath) {
+    return !pathResolver.isWithin('', dependencyPath);
+  }).toList();
+
+  if (pathDependencies.isNotEmpty) {
+    // Map the dependencies
+    final mappedDependencies = pathDependencies
+        .map(
+      (dependencyPath) => (
+        pathResolver.basename(dependencyPath),
+        dependencyPath,
+      ),
+    )
+        .fold(<String, String>{}, (map, dependency) {
+      map[dependency.$1] = dependency.$2;
+      return map;
+    });
+
+    final buildDirectory = Directory(
+      pathResolver.join(
+        directory.path,
+        'build',
+      ),
+    )..createSync();
+
+    // Create the packages directory
+    final packagesDirectory = Directory(
+      pathResolver.join(
+        buildDirectory.path,
+        '.dart_frog_path_dependencies',
+      ),
+    )..createSync();
+
+    // Copy all the dependencies to the packages directory
+    for (final entry in mappedDependencies.entries) {
+      final from = pathResolver.join(directory.path, entry.value);
+      final to = pathResolver.join(packagesDirectory.path, entry.key);
+
+      await copyPath(from, to);
+    }
```

Comment:
**Nit:** Simplify by using the already defined `pathDependencies`
```suggestion
    final dependencyOverrides = StringBuffer()
      ..writeln('dependency_overrides:');
    for (final dependencyPath in pathDependencies) {
      final basename = pathResolver.basename(dependencyPath);
      final from = pathResolver.join(directory.path, basename);
      final to = pathResolver.join(packagesDirectory.path, dependencyPath);

      await copyPath(from, to);

      dependencyOverrides.writeln('  $basename:\n    path: $to');
    }

    await File(
      pathResolver.join(
        buildDirectory.path,
        'pubspec_overrides.yaml',
      ),
    ).writeAsString(dependencyOverrides.toString());
```

Reply:
I would rather keep the operations separated, we definitely could put them together, but I feel the for would end up a little bit mixed in its responsabilities, and I would rather have the code easier to read.

If you feel too strong in the other direction, let me know and I can revisit.